bitnami/mongodb/templates/hidden/statefulset.yaml

```diff
@@ -123,51 +133,10 @@ spec:
             mountPath: /certs/CAs
           - name: certs
             mountPath: /certs
+          - name: common-scripts
+            mountPath: /bitnami/scripts
           command:
-            - sh
-            - "-c"
-            - |
-              /bin/bash <<'EOF'
-              my_hostname=$(hostname)
-              svc=$(echo -n "$my_hostname" | sed s/-[0-9]*$//)-headless
-              cp /certs/CAs/* /certs/
-              cat >/certs/openssl.cnf <<EOL
-              [req]
-              req_extensions = v3_req
-              distinguished_name = req_distinguished_name
-              [req_distinguished_name]
-              [ v3_req ]
-              basicConstraints = CA:FALSE
-              keyUsage = nonRepudiation, digitalSignature, keyEncipherment
-              subjectAltName = @alt_names
-              [alt_names]
-              DNS.1 = $svc
-              DNS.2 = $my_hostname
-              DNS.3 = $my_hostname.$svc.$MY_POD_NAMESPACE.svc.cluster.local
-              DNS.4 = localhost
-              DNS.5 = 127.0.0.1
-              {{- if .Values.externalAccess.hidden.service.loadBalancerIPs }}
-              {{- range $key, $val := .Values.externalAccess.hidden.service.loadBalancerIPs }}
-              IP.{{ $key }} = {{ $val | quote }}
-              {{- end }}
-              {{- else }}
-              IP.0 = {{ "$MY_POD_HOST_IP" | quote }}
-              {{- end }}
-              EOL
-
-              export RANDFILE=/certs/.rnd && openssl genrsa -out /certs/mongo.key 2048
-              #Create the client/server certificate
-              openssl req -new -key /certs/mongo.key -out /certs/mongo.csr -subj "/C=US/O=My Organisations/OU=IT/CN=$my_hostname" -config /certs/openssl.cnf
-              #Signing the server certificate with the CA cert and key
-              openssl x509 -req -in /certs/mongo.csr -CA /certs/mongodb-ca-cert -CAkey /certs/mongodb-ca-key -CAcreateserial -out /certs/mongo.crt -days 3650 -extensions v3_req -extfile /certs/openssl.cnf
-              rm /certs/mongo.csr
-              #Concatenate to a pem file for use as the client PEM file which can be used for both member and client authentication.
-              cat /certs/mongo.crt /certs/mongo.key > /certs/mongodb.pem
-              cd /certs/
-              shopt -s extglob
-              rm -rf !(mongodb-ca-cert|mongodb.pem|CAs|openssl.cnf)
-              chmod 0600 mongodb-ca-cert mongodb.pem
-              EOF
+            - /bitnami/scripts/generate-certs.sh
```

Comment:
I have noticed you use the same script `generate-certs.sh` for each role, but each one is slightly different.

Could be because some contributions forgot to modify them, or they have to be different. Have you checked it?

Reply:
Yes, main differences are about the DNS entries exposed. Maybe some combinations doesn't make sense (for example, arbiter certificate don't need the external names ). I can revisit it